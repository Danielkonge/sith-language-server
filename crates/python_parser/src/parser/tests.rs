mod context;
mod function;
mod invalid_assignments;
mod parser;
mod suite;
