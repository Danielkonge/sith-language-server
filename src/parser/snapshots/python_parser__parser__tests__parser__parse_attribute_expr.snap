---
source: src/parser/tests.rs
expression: "parse(\"\nvalue.attr\nvalue.attr()\nvalue().attr\nvalue().attr().foo\nvalue.attr.foo\n\")"
---
ParsedFile {
    ast: Module {
        body: [
            Expression(
                ExpressionStmt {
                    value: Attribute(
                        AttributeExpr {
                            value: Id(
                                Identifier {
                                    id: "value",
                                    range: 1..6,
                                },
                            ),
                            attr: Valid(
                                Identifier {
                                    id: "attr",
                                    range: 7..11,
                                },
                            ),
                            range: 1..11,
                        },
                    ),
                    range: 1..11,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Call(
                        CallExpr {
                            lhs: Attribute(
                                AttributeExpr {
                                    value: Id(
                                        Identifier {
                                            id: "value",
                                            range: 12..17,
                                        },
                                    ),
                                    attr: Valid(
                                        Identifier {
                                            id: "attr",
                                            range: 18..22,
                                        },
                                    ),
                                    range: 12..22,
                                },
                            ),
                            args: Arguments {
                                args: [],
                                kw_args: [],
                                range: 22..24,
                            },
                            range: 12..24,
                        },
                    ),
                    range: 12..24,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Attribute(
                        AttributeExpr {
                            value: Call(
                                CallExpr {
                                    lhs: Id(
                                        Identifier {
                                            id: "value",
                                            range: 25..30,
                                        },
                                    ),
                                    args: Arguments {
                                        args: [],
                                        kw_args: [],
                                        range: 30..32,
                                    },
                                    range: 25..32,
                                },
                            ),
                            attr: Valid(
                                Identifier {
                                    id: "attr",
                                    range: 33..37,
                                },
                            ),
                            range: 25..37,
                        },
                    ),
                    range: 25..37,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Attribute(
                        AttributeExpr {
                            value: Call(
                                CallExpr {
                                    lhs: Attribute(
                                        AttributeExpr {
                                            value: Call(
                                                CallExpr {
                                                    lhs: Id(
                                                        Identifier {
                                                            id: "value",
                                                            range: 38..43,
                                                        },
                                                    ),
                                                    args: Arguments {
                                                        args: [],
                                                        kw_args: [],
                                                        range: 43..45,
                                                    },
                                                    range: 38..45,
                                                },
                                            ),
                                            attr: Valid(
                                                Identifier {
                                                    id: "attr",
                                                    range: 46..50,
                                                },
                                            ),
                                            range: 38..50,
                                        },
                                    ),
                                    args: Arguments {
                                        args: [],
                                        kw_args: [],
                                        range: 50..52,
                                    },
                                    range: 38..52,
                                },
                            ),
                            attr: Valid(
                                Identifier {
                                    id: "foo",
                                    range: 53..56,
                                },
                            ),
                            range: 38..56,
                        },
                    ),
                    range: 38..56,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Attribute(
                        AttributeExpr {
                            value: Attribute(
                                AttributeExpr {
                                    value: Id(
                                        Identifier {
                                            id: "value",
                                            range: 57..62,
                                        },
                                    ),
                                    attr: Valid(
                                        Identifier {
                                            id: "attr",
                                            range: 63..67,
                                        },
                                    ),
                                    range: 57..67,
                                },
                            ),
                            attr: Valid(
                                Identifier {
                                    id: "foo",
                                    range: 68..71,
                                },
                            ),
                            range: 57..71,
                        },
                    ),
                    range: 57..71,
                },
            ),
        ],
        range: 0..71,
    },
    parse_errors: [],
}
